tools/clang/lib/CodeGen/ModuleBuilder.cpp

```diff
@@ -235,14 +236,16 @@ namespace {
                  end = Ctx.getSourceManager().fileinfo_end();
              it != end; ++it) {
           if (it->first->isValid() && !it->second->IsSystemFile) {
+            std::string filename = it->first->getName();
```

Comment:
I think it would be better if we could normalize paths earlier, so SourceManager paths are already good.  This ties into the string storage problem for filesMap as well.